R/metadata.R

```diff
@@ -74,13 +91,30 @@ imeta <- function(
   # define entity type
   entity_type <- match.arg(entity_type)
 
-  # check list depth if 1 add another layer
-  # TODO make a different check:
-  # operations should ALWAYS be a list of lists
-  # and those lists should have the right names inside.
-  # list of list => also a dataframe should be allowed
-  if (list_depth(operations) == 1)
-    operations <- list(operations)
+    # check for class `dataframe` and turn into list of lists
+  if (inherits(operations, "data.frame")) {
+    operations <- apply(operations, 1, as.list)
+  }
+
+  # check for names of `list` of lists
+  if (inherits(operations, "list")) {
+    content_operations <- unlist(operations, recursive = FALSE)
+    if (inherits(content_operations, "list")) {
+      operations_names <- unique(names(content_operations))
+      names_ref <- c("operation", "attribute", "value", "units")
+      if (!all(operations_names %in% names_ref)) {
+        names_msg <- paste0(paste0("\"", names_ref,  "\""), collapse = ", ")
+        stop("The supplied `operations` should have names that can include ",
+             names_msg, ".", call. = FALSE)
+      }
+    } else if (!is.null(content_operations)) {
+      stop("The supplied list of `operations` should contain a named `list`.",
+           call. = FALSE)
+    }
+  } else {
+    stop("The supplied `operations` should be of type `list` or `data.frame`.",
+         call. = FALSE)
+  }
```

Comment:
Should there also be a check of the possible values of "operation" ("add"/"remove")?

Reply:
Perhaps yes. Let me look at it later.